test/document_test.rb

```diff
@@ -443,6 +443,17 @@ def setup
     assert_equal [], second_doc.foo
   end
 
+  test 'attachments work properly' do
+    assert save_response = {ok: true, id: "base_dolly/79178957-96ff-40d9-9ecb-217fa35bdea7", rev: "1"}
+    assert FakeWeb.register_uri :put, /http:\/\/localhost:5984\/test\/base_dolly%2F.+/, body: save_response.to_json
+    assert doc = BaseDolly.new
+    assert doc.save
+    assert resp = {ok: true, id: '79178957-96ff-40d9-9ecb-217fa35bdea7', rev: '2'}
+    assert FakeWeb.register_uri :put, /http:\/\/localhost:5984\/test\/base_dolly\/79178957-96ff-40d9-9ecb-217fa35bdea7\/test.txt/, body: resp.to_json
+    assert data = File.open("#{FileUtils.pwd}/test/support/test.txt").read
+    assert doc.attach_file 'test.txt', 'text/plain', data
+  end
```

Comment:
Might be a good idea to include an assertion to verify the attachment is available in the document, even if it is a FakeWeb response.